pymc3/model.py

```diff
@@ -102,64 +103,70 @@ def get_named_nodes_and_relations(graph):
     graph - a theano node
 
     Returns:
-    leaf_nodes: A dictionary of name:node pairs, of the named nodes that
-        are also leafs of the graph
-    node_parents: A dictionary of node:set([parents]) pairs. Each key is
+    leafs: A dictionary of name:node pairs, of the named nodes that
+        have no named ancestors in the provided theano graph.
+    descendents: A dictionary of node:set([parents]) pairs. Each key is
         a theano named node, and the corresponding value is the set of
-        theano named nodes that are parents of the node. These parental
-        relations skip unnamed intermediate nodes.
-    node_children: A dictionary of node:set([children]) pairs. Each key
+        theano named nodes that are direct descendents of the node in the
+        supplied ``graph``. These relations skip unnamed intermediate nodes.
+    ancestors: A dictionary of node:set([ancestors]) pairs. Each key
         is a theano named node, and the corresponding value is the set
-        of theano named nodes that are children of the node. These child
-        relations skip unnamed intermediate nodes.
+        of theano named nodes that are direct ancestors in the of the node in
+        the supplied ``graph``. These relations skip unnamed intermediate
+        nodes.
 
     """
     if graph.name is not None:
-        node_parents = {graph: set()}
-        node_children = {graph: set()}
+        ancestors = {graph: set()}
+        descendents = {graph: set()}
     else:
-        node_parents = {}
-        node_children = {}
-    return _get_named_nodes_and_relations(graph, None, {}, node_parents, node_children)
-
-def _get_named_nodes_and_relations(graph, parent, leaf_nodes,
-                                        node_parents, node_children):
+        ancestors = {}
+        descendents = {}
+    descendents, ancestors = _get_named_nodes_and_relations(
+        graph, None, ancestors, descendents
+    )
+    leafs = {
+        node.name: node for node, ancestor in ancestors.items()
+        if len(ancestor) == 0
```

Comment:
I think pylint dislikes testing for length = 0, but to be honest, I don't know why.

Reply:
I think that I'll leave this as is, given that our pylint tests are passing. I don't want to rely on the implicit truthfulness of an empty set.

Reply:
Agreed. I think pylint is wrong about that, especially since len works for things that truthiness doesn't.